presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -174,15 +178,20 @@ private DistributedQueryRunner(
             extraCoordinatorProperties.put("experimental.iterative-optimizer-enabled", "true");
             extraCoordinatorProperties.putAll(extraProperties);
             extraCoordinatorProperties.putAll(coordinatorProperties);
-            coordinator = closer.register(createTestingPrestoServer(discoveryUrl, false, resourceManagerEnabled, true, extraCoordinatorProperties, parserOptions, environment, baseDataDir));
-            servers.add(coordinator);
+            for (int i = 0; i < coordinatorCount; i++) {
+                TestingPrestoServer coordinator = closer.register(createTestingPrestoServer(discoveryUrl, false, resourceManagerEnabled, true, extraCoordinatorProperties, parserOptions, environment, baseDataDir));
+                servers.add(coordinator);
+                coordinators.add(coordinator);
+                extraCoordinatorProperties.remove("http-server.http.port");
+            }
 
             if (resourceManagerEnabled) {
-                resourceManager = Optional.of(closer.register(createTestingPrestoServer(discoveryServer.getBaseUrl(), true, resourceManagerEnabled, false, extraCoordinatorProperties, parserOptions, environment, baseDataDir)));
+                resourceManager = Optional.of(closer.register(createTestingPrestoServer(discoveryUrl, true, resourceManagerEnabled, false, extraCoordinatorProperties, parserOptions, environment, baseDataDir)));
```

Comment:
ditto, we should pass new http-server.http.port value for resource manager to make it explicit. For now i believe it runs on port 0 given we have removed the property when configuring coordinator.

Reply:
The other coordinators will all receive a port 0, which means pick any available port.  It works on >2 coordinators (you'll see from the logs that each additional coordinator is assigned a port at random, similar to the workers).